chainer/training/extensions/evaluator.py

```diff
@@ -0,0 +1,168 @@
+import copy
+
+import six
+
+from chainer.dataset import convert
+from chainer.dataset import iterator as iterator_module
+from chainer import link
+from chainer import reporter as reporter_module
+from chainer.training import extension
+from chainer import variable
+
+
+class Evaluator(extension.Extension):
+
+    """Trainer extension to evaluate models on a validation set.
+
+    This extension evaluates the current models by a given evaluation function.
+    It creates a :class:`~chainer.Reporter` object to store values observed in
+    the evaluation function on each iteration. The report for all iterations
```

Comment:
I think we should refer to [report's document](https://github.com/pfnet/chainer/pull/1285/files#diff-c085f5638262ee03f0c905fe364e5704R190) for naming rule of observed values.